src/commands/stac-validate/stac.validate.ts

```diff
@@ -213,6 +207,105 @@ export const commandStacValidate = command({
   },
 });
 
+/**
+ * Validate STAC Assets
+ *
+ * @param stacJson
+ * @param path absolute path of the STAC location
+ * @returns the list of link paths that failed the validation
+ */
+export async function validateAssets(
+  stacJson: st.StacItem | st.StacCollection | st.StacCatalog,
+  path: string,
+): Promise<string[]> {
+  const assetsFailures: string[] = [];
+  const assets = Object.values(stacJson.assets ?? {}) as st.StacAsset[];
+  for (const asset of assets) {
+    const isChecksumValid = await validateStacChecksum(asset, path, { allowMissing: false, allowUnknown: false });
+    if (!isChecksumValid) {
+      assetsFailures.push(path);
+    }
+  }
+  return assetsFailures;
+}
+
+/**
+ * Validate STAC Links
+ *
+ * @param stacJson
+ * @param path absolute path to the STAC location
+ * @returns the list of link paths that failed the validation
+ */
+export async function validateLinks(
+  stacJson: st.StacItem | st.StacCollection | st.StacCatalog,
+  path: string,
+): Promise<string[]> {
+  const linksFailures: string[] = [];
+  for (const link of stacJson.links) {
+    if (link.rel === 'self') continue;
+    // we `allowMissing` because some STAC links might not have a checksum yet (feature added later)
+    const isChecksumValid = await validateStacChecksum(link, path, { allowMissing: true, allowUnknown: false });
+    if (!isChecksumValid) {
+      linksFailures.push(path);
+    }
+  }
+  return linksFailures;
+}
+
+/**
+ * Configuration to validate checksums within a STAC object
+ */
+interface ValidateStacChecksumContext {
+  /**
+   * Not valid if checksum is missing
+   */
+  allowMissing: boolean;
+  /**
+   * Not valid if checksum type (different than SHA256) is unknown
+   */
+  allowUnknown: boolean;
+}
+
+/**
+ * Validate if the checksum found in the stacObject (`file:checksum`) corresponds to its actual file checksum.
+ * @param stacObject
+ * @param path
+ * @param ctx
+ * @returns
+ */
+export async function validateStacChecksum(
+  stacObject: st.StacLink | st.StacAsset,
+  path: string,
+  ctx: ValidateStacChecksumContext,
+): Promise<boolean> {
+  let source = stacObject.href;
+  if (source.startsWith('./')) source = fsa.join(dirname(path), source.replace('./', ''));
+  const checksum: string = stacObject['file:checksum'] as string;
+
+  if (checksum == null) {
+    if (ctx.allowMissing) return true;
+    logger.error({ source, checksum }, 'Validate:Checksum:Missing');
+    return false;
+  }
+  // 12-20 is the starting prefix for all sha256 multihashes
+  if (!checksum.startsWith('1220')) {
+    if (ctx.allowUnknown) return true;
```

Comment:
Do we want to be able to pass `allowUnknown = true`?

Reply:
Probably not at this stage